src/DocBlock/TypeExpression.php

```diff
@@ -208,8 +208,9 @@ final class TypeExpression
 
     private string $value;
 
-    private bool $isUnionType;
+    private bool $isGluedType;
 
+    /** @var '&'|'|'|'~' */
```

Comment:
I am not sure if subtraction types are supported already (phpstan/phpstan#3182 is still open). Personally I would not mix clean up around union/intersection with introducing support for `~` (s). Can we drop subtraction types from this PR? I feel comfortable with merging refactor around union/intersection types, new composite type should be discussed with Keradus.

Reply:
You are right, they are supported in phpstan only internally, but not parseable yet. I will remove the support from this PR.